### DISCUSSION TITLE
Supplemental Guidance Grouping

### DISCUSSION THREAD

**Discussion by ChrisLoiselle** [w3c/low-vision-a11y-tf#125] [Ideas]
Perhaps grouping at a high level, https://html.spec.whatwg.org/multipage/dom.html#concept-html-contents would be beneficial in addition to how we are currently iterating on the table design. Just a thought worth discussing on another call to get all involved in same mind set, as we move forward as a group.

**Comment by mraccess77** [comment]
Below is a link to the Google Sheets version of the Supplemental Guidance list.

https://docs.google.com/spreadsheets/d/1PNaDkFNhANt7T1IPcyfx7uwPXeB8eE5EFtoXjqexEvg/edit?usp=sharing

**Reply by mraccess77** [reply]
Andy's proposed grouping (as I understand it) https://github.com/w3c/low-vision-a11y-tf/discussions/129

- Fluent Readability, Block/Body Text
- Fluent Readability, Primary content that is not body text
- Spot Readability, secondary and ancillary "non-content"
- Non-Text Discernible Elements
- Non-Text Functional Elements
- Aesthetic and Divisional Elements
- Aesthetic and Divisional Elements SHALL NOT distract nor interfere with readable content.

### Sam's proposal for grouping:
### Visual appearance of body paragraphs of text (Body text defined as more than two lines of related text in a block or column.)
- Size of body text	For body paragraphs of text, avoid large text and avoid small text. Font-size is not standardised across different fonts, so this will need to be specified in terms of x-height, and translated into what this means for different fonts. For a well-behaved font like Helvetica, it will probably end up something like font-size should be at least 12px and at most 18 px (TBC)
- Lightness Contrast of body text	For body paragraphs of text, avoid low contrast text and avoid high contrast text. Large text can be readable at low contrasts, but large text is not practical for body paragraphs of text, hence low contrast text should be avoided. High contrast text can cause issues with migraines (TBC). Exact values TBC, depending on whether we do it as a ratio or APCA contrast di
- Font-weight of body text	For body paragraphs of text, avoid thin fonts, and avoid thick fonts. Font-weight is not standardised across fonts so this may need to be specified in terms of the ratio of the stroke width compared to the x height, or by describing the limits in terms of some reference fonts that are well behaved. For a well-behaved font like Helvetica, font-weight for continuous paragraph
- Padding and line spacing for body text	Ensure sufficient padding between text and the edge of the viewport / any other elements
ensure sufficient padding between each paragraph, this should be visually distinct from the line spacing within paragraphs.
Line-height is not standardised across different fonts. A font like verdana needs a much bigger line height than most other fonts. This may need 

### Visual appearance of text for spot reading (individual words or single lines)
- Size, lightness contrast and font-weight of text for spot reading. For text for spot reading, it's not feasible to set thresholds for size, contrast and font weight that are independent of each other. The APCA tool attempts to address this, further work is needed to verify some simplified minimum thresholds for size, contrast and font weight. Nonuniformity of font-weight and font-size for different
- Padding and line spacing for body text	Single lines of text may still become multiple lines when the text is zoomed, so line-height is still an issue for single lines of text (and often forgotten!). However the line-height probably doesn't have to be as big as the body paragraphs of text (TBC)
Background colour behind the text must continue at least 1em round the text in all directions.

**Reply by sdw32** [reply]
Indeed, that seems like a reasonable summary. I think the primary decision here is whether we split into 2 categories for the different kinds of text content, or 3 categories. In one of the recent references I have read, fluent reading is defined as greater than 80 words per minute, and spot reading is defined as greater than 40 words per minute. 

So, this leads to a particularly pertinent question: For headings, navigation menu text, text on buttons, and keywords within diagrams, is spot reading at 40 words per minute sufficient?  If 40wpm is not sufficient, then we might need three categories as Andy suggests. If 40wpm is sufficient, then perhaps we could manage with two categories.

If we do need 3 categories, I think we would need a robust, and preferably automated method that could be used to determine whether text is  'primary content that is not body text' or 'secondary content'. 

If the guidance would be the same for 'primary content that is not body text' and 'secondary content', then I think it would be more straightforward to write, and shorter, and easier to apply/understand.

I will look forward to further discussions on this topic, many thanks.

**Comment by mraccess77** [comment]
Bhoomika and Sam comment: grouping, matrix, or tags might be useful to show theme.  Group items related to zooming.
